src/commands/fieldModifier.ts

```diff
@@ -33,110 +34,82 @@ function buildAndOpenModal(
     }
 }
 
-function createDvField(
+export async function fieldModifier(
     plugin: MetadataMenu,
     dv: any,
     p: any,
-    fieldContainer: HTMLElement,
     fieldName: string,
     attrs?: { cls?: string, attr?: Record<string, string>, options?: Record<string, string> }
-): void {
-    const field = plugin.fieldIndex.filesFields.get(p.file.path)?.filter(f => f.isRoot()).find(field => field.name === fieldName)
-    if (!field?.isRoot()) {
-        /*
-        field modifiers are only available for root fields
-        */
-        dv.el('span', p[field!.name], attrs);
-        return
-    }
-    if (field?.type) {
-        const target = plugin.app.vault.getAbstractFileByPath(p.file.path) as TFile
-        const fieldVM = fieldValueManager(plugin, field.id, field.fileClassName, target, undefined)
-        _createDvField(fieldVM, dv, p, fieldContainer)
-    } else {
-        const field = buildField(plugin, fieldName, "", "", undefined, undefined, undefined, undefined, "Input", {});
-        const file = plugin.app.vault.getAbstractFileByPath(p.file.path)
-        if (file instanceof TFile) {
-            const fieldVM = new (FieldValueManager(plugin, field, file, undefined))
-            _createDvField(fieldVM, dv, p, fieldContainer, attrs)
-        }
-    }
-}
-
-export function fieldModifier(
-    plugin: MetadataMenu,
-    dv: any,
-    p: any,
-    fieldName: string,
-    attrs?: { cls?: string, attr?: Record<string, string>, options?: Record<string, string> }
-): HTMLElement {
+): Promise<HTMLElement> {
     /* fieldContainer*/
     const fieldContainer: HTMLElement = dv.el("div", "")
     fieldContainer.setAttr("class", `metadata-menu-dv-field-container ${fieldName}`)
 
     /* create fieldModifier depending on fileClass type or preset value*/
-    if (p[fieldName] === undefined) {
+    const file = plugin.app.vault.getAbstractFileByPath(p.file.path)
+    if (!(file instanceof TFile && file.extension == "md")) {
+        throw Error("path doesn't correspond to a proper file");
+    }
+    const { indexedPath, field } = getFullIndexedPath(fieldName, plugin.fieldIndex.filesFields.get(file.path))
+    const ef = await getExistingFieldForIndexedPath(plugin, file, indexedPath)
```

Comment:
Is it a problem to use an async function here?

Reply:
There are 2 drawbacks using this approach (getting the managedField.value from ExisitingField in a fieldModifier and therefore convert it to an async function):
- it requires to rebuild a Note for each field in the table, whereas the value of the field is already indexed by dataview => that leads to extra CPU + RAM usage and slows down the building of the table
- it requires extra code to await for every fieldModifier to be resolved before building the table. It could be Ok in the fileclass tableview and mdm codeblock because this `await Promise.all([...].map(async p => await fieldModifier(....))` but some users are building their own dataviewjs queries with fieldModifier applied to certain fields and this syntax is very complicated to understand IMHO.

Reply:
Gotcha! Let me take another swing then!

Reply:
Swapped to grabbing the value from the page object directly upfront and setting the field manager value before creating the DV element! Alternatively It could be passed to the constructor but I'm not sure which is preferred!

Reply:
Bump! Would love to get feedback/merge so I can retire my fork :D